theano/sandbox/gpuarray/elemwise.py

```diff
@@ -101,31 +101,55 @@ def make_node(self, *inputs):
     def generate_kernel(self, node, nodename):
         inps = [make_argument(i, 'i%d' % (n,)) for n, i in
                 enumerate(node.inputs)]
-        scal_ins = [scalar.get_scalar_type(i.dtype) for i in node.inputs]
+        scal_v_ins = [scalar.get_scalar_type(i.dtype) for i in node.inputs]
 
         outs = [make_argument(o, 'o%d' % (n,)) for n, o in
                 enumerate(node.outputs) if not n in self.inplace_pattern]
-        scal_out = [scalar.get_scalar_type(o.dtype) for o in node.outputs]
+        scal_v_outs = [scalar.get_scalar_type(o.dtype) for o in node.outputs]
 
-        fake_node = Apply(self.scalar_op, [i() for i in scal_ins],
-                          [o() for o in scal_out])
+        fake_node = Apply(self.scalar_op, [i() for i in scal_v_ins],
+                          [o() for o in scal_v_outs])
+
+        scal_in = [i.name + '[i]' if i.dtype != 'float16' else
+                   '__half2float(' + i.name + '[i])' for i in inps]
```

Comment:
That's "special" since I have to do some trickery for float16 anyway.  So I don't need to use the stuff from fp16_help.